packages/resolve-command/src/index.js

```diff
@@ -7,22 +8,72 @@ const verifyCommand = async ({ aggregateId, aggregateName, type }) => {
   if (!type) throw new Error('The "type" argument is required')
 }
 
+const emptySnapshotAdapter = Object.freeze({
+  loadSnapshot: async () => ({ timestamp: 0, state: null, version: 0 }),
+  saveSnapshot: () => null
+})
+
+const makeCommandHandlerHash = (projection, aggregateId) =>
+  Object.keys(projection)
+    .sort()
+    .map(
+      handlerName =>
+        `${handlerName}:${invariantFunctionHash(projection[handlerName])}`
+    )
+    .join(',') + `;${aggregateId}`
+
 const getAggregateState = async (
-  { projection, initialState },
+  {
+    projection,
+    initialState,
+    snapshotAdapter = emptySnapshotAdapter,
+    snapshotBucketSize = 100
+  },
   aggregateId,
   eventStore
 ) => {
-  const handlers = projection || {}
-  let aggregateState = initialState
+  const snapshotKey =
+    projection && projection.constructor === Object
+      ? makeCommandHandlerHash(projection, aggregateId)
+      : null
+
+  let aggregateState = null
   let aggregateVersion = 0
+  let appliedEvents = 0
+  let lastTimestamp = 0
 
-  await eventStore.getEventsByAggregateId(aggregateId, event => {
-    aggregateVersion = event.aggregateVersion
-    const handler = handlers[event.type]
-    if (!handler) return
+  try {
+    if (snapshotKey == null) throw new Error()
+    const snapshot = await snapshotAdapter.loadSnapshot(snapshotKey)
+    aggregateVersion = snapshot.version
+    aggregateState = snapshot.state
+    lastTimestamp = snapshot.timestamp
+  } catch (err) {}
+
+  if (!(+lastTimestamp > 0)) {
+    aggregateState = initialState
+  }
 
-    aggregateState = handler(aggregateState, event)
-  })
+  await eventStore.getEventsByAggregateId(
+    aggregateId,
+    event => {
+      aggregateVersion = event.aggregateVersion
+      const handler = projection && projection[event.type]
+      if (typeof handler !== 'function') return
+
+      aggregateState = handler(aggregateState, event)
+
+      if (snapshotKey != null && ++appliedEvents % snapshotBucketSize === 0) {
```

Comment:
add brackets 
snapshotKey != null && (++appliedEvents % snapshotBucketSize === 0)
or even
(snapshotKey != null) && ((++appliedEvents % snapshotBucketSize) === 0)

Reply:
It's [https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKAzlAhgB3QCwhgGk4BPAAgEIBeCqAVwBsmBKCgMg4tVQGo+uHEwCWcACYBRAG4IY6CgFIKmXASIAhBmADWcGAGURALzjsaFigAZWAHSj37qvIRLlqdRi07d+gnMJiUrKwCsrO6jBauvpGphQWdDb2IAA0IBA4MCLQ6MigWABOhRAA7gAKRQh5KFjSECLiaSBY6DDIAGZYTOhw6QBGhVgxhjjDIlAA5sgwhQx9IOIQYJ3dvekTvYUw5UOTALZYqz0LAFboAB4aQyMGWPtwADITcMfrGQwwOJ8ATG8LY0KW2QIH6WH6ZCY0GaOEKExgAHVGjB8MgABxWdKwiC9BFDHAg2FwLayZqFOAARwYInJuywByOSC6J3SvX2IhmcwW6AmkyYcAAigwiK8mWsFjBwUjxCjkD90rMsCJRFMAMIQfaHEFQaCvdIMXoAFXBNWZvQAvuagA](`prettier` work)